spec/models/category_list_spec.rb

```diff
@@ -116,7 +116,12 @@
   end
 
   describe 'category order' do
-    let(:category_ids) { CategoryList.new(Guardian.new(admin)).categories.map(&:id) - [SiteSetting.uncategorized_category_id] }
+    def category_list(user)
+      CategoryList.new(Guardian.new(user)).categories.map(&:id) - [SiteSetting.uncategorized_category_id]
```

Comment:
I worry that using category list here is too expensive, it does a lot of work including pruning empty dealing with unpinned per category and so on. We end up throwing all this work away.